react/src/components/search-app/SearchBarComponent.jsx

```diff
@@ -6,7 +6,7 @@ export default class SearchBarComponent extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      query: props.query,
+      query: props.initQuery,
```

Comment:
Good, `initQuery` makes it clear it's just for initialization. I think that's acceptable 👍